app/services/invoices/create_service.rb

```diff
@@ -32,12 +32,13 @@ def create
           timestamp: timestamp,
         ).call
 
-        SendWebhookJob.perform_later(:invoice, invoice) if should_deliver_webhook?
         Invoices::Payments::CreateService.new(invoice).call
         track_invoice_created(invoice)
 
         result
       end
+
+      SendWebhookJob.perform_later(:invoice, invoice) if should_deliver_webhook?
```

Comment:
I think both `Invoices::Payments::CreateService.new(invoice).call` and `track_invoice_created(invoice)` are in the same situation

And we should keep returning the `result` :)

Reply:
oh yes sorry for the `result`, I'll move the other 2 also